devito/dse/inspection.py

```diff
@@ -21,7 +21,10 @@ def terminals(expr, discard_indexed=False):
 
     # Use '.name' for quickly checking uniqueness
     junk = flatten([i.free_symbols for i in indexed])
-    junk = [i.name for i in junk]
+    junk = [i.name for i in junk] + \
+           [i.parent.name if
+            (i.is_Buffered if hasattr(i, "is_Buffered") else False)
```

Comment:
``... + [i.parent.name for i in junk if isinstance(i, BufferedDimension)]``

Reply:
Thanks

Just not sure if this should be here. I hacked it because it work but seems odd and out of place.